internal/state/validation_test.go

```diff
@@ -126,11 +126,6 @@ func TestValidateBlockHeader(t *testing.T) {
 			"Proposer invalid",
 			func(block *types.Block) { block.ProposerProTxHash = []byte("wrong size") },
 		},
-		// Set appVersion to 2 allow "invalid proposed app version" case
```

Comment:
Why do we removed this test?

Reply:
As we introduced ResponsePrepareProposal.app_version , we need to allow state.Version.Consensus.App to be incorrect (because it will be overridden anyway). So relevant check was removed from validation.go:31 and is removed also here.